File: ischia/src/test/java/cz/fb/manaus/ischia/LayLoserBettorTest.java
package cz.fb.manaus.ischia;

import cz.fb.manaus.core.model.MarketPrices;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.reactor.betting.AbstractBettorTest;
import org.junit.Test;
import org.springframework.test.context.ActiveProfiles;

import java.util.OptionalDouble;

import static java.util.OptionalDouble.of;

@ActiveProfiles("ischia")
public class LayLoserBettorTest extends AbstractBettorTest<LayLoserBettor> {

    @Test
    public void testPlaceFairness() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.98d, 3.2d, of(3.05d), 1)),
                3, OptionalDouble.of(2.88));
    }

    @Test
    public void testPlaceBest() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.58d, 3.15d, of(3d), 1)),
                3, OptionalDouble.of(2.6d));
    }

    @Test
    public void testLastMatchedOrTradedVolume() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.8d, 3.2d, of(2.5), 1)),
                3, OptionalDouble.of(2.48));
    }

    @Test
    public void testCloseUpdate() {
        MarketPrices market = persistMarket(reactorTestFactory.createMarket(2.90d, 3.2d, of(3d), 1));

        checkUpdate(market, 2.86, Side.LAY, 0, 0);
        checkUpdate(market, 2.88, Side.LAY, 0, 0);

        checkUpdate(market, 2.9, Side.LAY, 0, 3);
        checkUpdate(market, 2.84, Side.LAY, 0, 3);
        checkUpdate(market, 2.92, Side.LAY, 0, 3);
        checkUpdate(market, 2.94, Side.LAY, 0, 3);
    }

}

File: ischia/src/test/java/cz/fb/manaus/ischia/LayLoserBettorTest.kt
package cz.fb.manaus.ischia

import cz.fb.manaus.core.model.Side
import cz.fb.manaus.core.test.AbstractIntegrationTestCase
import cz.fb.manaus.reactor.PricesTestFactory
import cz.fb.manaus.reactor.betting.BettorTester
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ActiveProfiles

@ActiveProfiles("ischia")
class LayLoserBettorTest : AbstractIntegrationTestCase() {

    @Autowired
    private lateinit var bettorTester: BettorTester


    @Test
    fun `place bet - based on fairness`() {
        val marketPrices = PricesTestFactory.newMarketPrices(2.98, 3.2, 3.05)
        bettorTester.checkPlace(Side.LAY, marketPrices, 3, 2.84)
    }

    @Test
    fun `place bet - based on best price`() {
        val marketPrices = PricesTestFactory.newMarketPrices(2.58, 3.15, 3.0)
        bettorTester.checkPlace(Side.LAY, marketPrices, 3, 2.596)
    }

    @Test
    fun `upadte bet`() {
        val market = PricesTestFactory.newMarketPrices(2.90, 3.2, 3.0)
        bettorTester.checkUpdate(Side.LAY, 3.0, market, 0, 3)
    }

}
